tests/Craftify.Geometry.Domain.Tests/LineSegmentsGraphAlgorithmsTests.cs

```diff
@@ -0,0 +1,117 @@
+        public static LineSegment ThirdSegment => Line.ByStartPointAndEndPoint(
+                Point.ByCoordinates(1.22124532708768E-15, 5.00000000000004, 0),
+                Point.ByCoordinates(5.00000000000005, 5.00000000000006, 0));
+        public static LineSegment FourthSegment => Line.ByStartPointAndEndPoint(
+                Point.ByCoordinates(5.00000000000007, 5.00000000000008, 0),
+                Point.ByCoordinates(5.00000000000009, 10.00000000000001, 0));
+        public static LineSegment FifthSegment => Line.ByStartPointAndEndPoint(
+                Point.ByCoordinates(-5.00000000000002, 5.00000000000003, 0),
+                Point.ByCoordinates(-5.00000000000004, 10.00000000000005, 0));
+        public static LineSegment SixthSegment => Line.ByStartPointAndEndPoint(
+                Point.ByCoordinates(5.00000000000006, 10.00000000000007, 0),
+                Point.ByCoordinates(0.00000000000008, 10.00000000000009, 0));
+        public static LineSegment SeventhSegment => Line.ByStartPointAndEndPoint(
+                Point.ByCoordinates(-5.00000000000001, 10.00000000000002, 0),
+                Point.ByCoordinates(0.00000000000003, 10.00000000000004, 0));
+        public static LineSegment EigthSegment => Line.ByStartPointAndEndPoint(
+                Point.ByCoordinates(0.00000000000005, 10.00000000000006, 0),
+                Point.ByCoordinates(0.00000000000007, 15.00000000000008, 0));
+    }
+
+    static LineSegmentsGraph _graph => new(ReadOnlyList.Of(
+        [
+            GraphData.FirstSegment,
+            GraphData.SecondSegment,
+            GraphData.ThirdSegment,
+            GraphData.FourthSegment,
+            GraphData.FifthSegment,
+            GraphData.SixthSegment,
+            GraphData.SeventhSegment,
+            GraphData.EigthSegment
+        ]
+    ));
+
+    [Theory]
+    [MemberData(nameof(DijkstraTestData))]
+    public void Dijkstra_ShouldReturnShortestPath(
+        LineSegmentsGraph graph,
+        Point3D start,
+        Point3D end,
+        IReadOnlyList<LineSegment> expected)
+    {
+        var actualPath = graph.Dijkstra(start, end).ToList();
+
+        if (!expected.Any())
+        {
+            actualPath.Should().BeEmpty();
+        }
+        else
+        {
+            actualPath
+                .Should()
+                .HaveSameCount(expected)
+                .And
+                .AllSatisfy(edge =>
+                {
+                    int index = actualPath.IndexOf(edge);
+
+                    edge.Start.AlmostEqualTo(expected[index].Start);
+                    edge.End.AlmostEqualTo(expected[index].End);
+                });
+        }
+    }
+
+    public static IEnumerable<object[]> DijkstraTestData()
+    {
+        yield return [
+            _graph,
+            Point.ByCoordinates(1, 1, 1),
+            Point.Origin(),
+            ReadOnlyList.Of<LineSegment>()
+        ];
+
+        yield return [
+            _graph,
```

Comment:
Let's try to create as few abstractions as possible. In this case, we store `_graph` as a static getter. What value does it bring? Do we reuse that graph somewhere else?

To understand the test, I need to take the following steps:

1. Find `Dijkstra_ShouldReturnShortestPath`.
2. Understand what graph is passed to the parameter.
3. Find the method `DijkstraTestData`, which passes this graph.
4. Then, again, this graph is set in another location.

This doesn't bring any value.

To me, simply creating the graph directly in `Dijkstra_ShouldReturnShortestPath` makes it much more readable. I find the test `Dijkstra_ShouldReturnShortestPath`, see what data we pass, and notice that the start and end values vary. I don't need to jump through abstractions (methods in this case).